### PR TITLE
Read osi format

### PR DESCRIPTION
**Pull request by vkresch**
#### Reference to a related issue in the repository
The [PR](https://github.com/OpenSimulationInterface/open-simulation-interface/pull/358) which defines the new OSI file format and its benefits.

#### Add a description
**What is this change?**
This PR adds the feature for the osi visualizer to read and record OSI trace files in  the new osi file format which is length separated.

**Is this a bug fix or a feature? Does it break any existing functionality or force me to update to a new version?**
This PR is a feature and does not break any functionality and is still backwards compatible since the old txt separated by $$__$$ trace files can still be read. The recording will always output the OSI trace format to motivate user to use the new file format. 

**How has it been tested?**
The functionality has been tested with multiple txt and osi trace files. The output of the recording has been used again as a input to see if the visualizer recognizes the format.

**Change notes:**
- Extended input files to *.txt and *.osi
- Added two functions (`read_bytes`, `realloc_buffer`) for reading and allocating in the new file format
- Recorder export outputs now **only** to *.osi
- Removed OSMP from gitmodule
- Simplyfiction of some expression like `if(var==true)` -->` if(var)`
- Changed nanoseconds and slidervalue to uint64_t
- replaced `push_back(std::make_pair(` with `emplace_back(`

#### Mention a member
@jdsika @pmai let me know your thoughts, suggestions and improvements! Thanks ;)

#### Check the checklist

- [x] I have performed a self-review of my own code.
- [x] I have made corresponding changes to the [documentation](https://github.com/OpenSimulationInterface/osi-documentation).
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests / travis ci pass locally with my changes.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -1,7 +1,3 @@
 [submodule "open-simulation-interface"]
 	path = open-simulation-interface
 	url = https://github.com/OpenSimulationInterface/open-simulation-interface.git
-
-[submodule "osi-sensor-model-packaging"]
-	path = osi-sensor-model-packaging
-	url = https://github.com/OpenSimulationInterface/osi-sensor-model-packaging.git
```
Comment (jdsika) [inline .gitmodules]: Was this never used?

File: include/osireader.h [hunk 2]
```diff
@@ -26,7 +26,7 @@ class OsiReader : public QObject, public IMessageSource
     Q_OBJECT
 
   public:
-    OsiReader(int* deltaDelay, const bool& enableSendOut, const std::string& pubPortNum, int socketType);
+    OsiReader(const int* deltaDelay, const bool& enableSendOut, std::string  pubPortNum, int socketType);
```
Comment (jdsika) [inline include/osireader.h]: Why do you work with pointers here? I think it is way better to use references? pubPostNum is not const anymore - why did you remove the reference? Only for int I think the reference does not save memory (same size pointer and int)

File: .gitmodules (hunk 1 above)
Comment (vkresch) [reply .gitmodules]: Not sure I mean we don't have it as a subfolder now and for installation it is not needed.

File: include/osireader.h (hunk 2 above)
Comment (vkresch) [reply include/osireader.h]: clang tidy suggested me the changes :)
